### PR TITLE
HPCC-15134 Ensure parallel activities are not executed inline

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@shamser please review

File: ecl/hql/hqlfold.cpp [hunk 1]
```diff
@@ -2447,7 +2447,7 @@ IHqlExpression * foldConstantOperator(IHqlExpression * expr, unsigned foldOption
         {
             IHqlExpression * child = expr->queryChild(0);
             IValue * constValue = child->queryValue();
-            unsigned last = expr->numChildren()-1;
+            unsigned last = numNonAttributes(expr)-1;
```
Comment (shamser) [inline ecl/hql/hqlfold.cpp]: numNonAttribute may return 0, so that (unsigned) last is assigned -1.

**Comment by shamser** [comment]
@ghalliday One minor comment.

File: ecl/hql/hqlfold.cpp (hunk 1 above)
Comment (ghalliday) [inline ecl/hql/hqlfold.cpp]: Theoretically true, but that would require an invalid IHqlExpression - no_choose must have at least 2 non-attribute expressions.

**Pull request merged**